test/new-e2e/tests/containers/ecs_test.go

```diff
@@ -73,6 +78,93 @@ type ecsSuite struct {
 	ecsClusterName string
 }
 
+func (suite *ecsSuite) TestAgent() {
+	ctx := context.Background()
+
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
+	suite.Require().NoErrorf(err, "Failed to load AWS config")
+
+	client := awsecs.NewFromConfig(cfg)
+
+	suite.Run("ECS tasks are ready", func() {
```

Comment:
a newbie question: since the execution order is not sequential, how do we guarantee this test is executed before other tests?

Reply:
Good point!
Inside a testify test suite, tests are executed in alphabetical order. As `TestAgent` comes before `TestDogstatsd`, `TestNginx`, etc., it is executed first.
That’s definitively fragile as it’s not very explicit.

I can rename the function `Test00Agent` and add a comment explaining that the `00` is here to guarantee that this tests comes first in the alphabetical order.

Reply:
Can we move it to SetupTest/Suite?

Reply:
@zhuminyi : I’ve added https://github.com/DataDog/datadog-agent/pull/20332/commits/ab8f3c252a45dbd5500c75ce58d8efde199e3550 to document why and how the test waiting for pods or ECS tasks to be ready must be run first.